locator/azure_snitch.cc

```diff
@@ -44,7 +44,8 @@ future<> azure_snitch::load_config() {
 
     logger().info("AzureSnitch using region: {}, zone: {}.", azure_region, azure_zone);
 
-    _my_rack = azure_zone;
+    // Zoneless regions return empty zone
+    _my_rack = (azure_zone != "" ? azure_zone : azure_region);
```

Comment:
Is it correct to set the region as the zone?

Reply:
It's just an identifier checked later against equal/not-equal against over rack ~~strings~~ identifiers. If zone is not returned, it means all nodes in the cluster (should) live in one region-wide zone, so it can be anything

Reply:
@avikivity , do you still have concerns about it?